examples/apps/echo/client/echo_client.py

```diff
@@ -166,36 +172,61 @@ def Main(args=None):
 	requester_id = secrets.token_hex(32)
 	session_iv = utility.generate_iv()
 	session_key = utility.generate_key()
-	encrypted_session_key = utility.generate_encrypted_key(
-		session_key, worker_obj.encryption_key)
-	requester_nonce = secrets.token_hex(32)
+	requester_nonce = secrets.token_hex(16)
 	# Create work order
-	wo_submit_json = jrpc_request.WorkOrderSubmitJson(3, 6000, "JSON-RPC",
+	wo_submit_json = WorkOrderParams(
 		work_order_id, worker_id, workload_id, requester_id,
-		encrypted_session_key, session_iv, requester_nonce,
-		worker_encryption_key=base64.b64decode(
-			worker_obj.encryption_key).hex(),
+		session_key, session_iv, requester_nonce,
+		result_uri=" ", notify_uri=" ",
+		worker_encryption_key=worker_obj.encryption_key,
 		data_encryption_algorithm="AES-GCM-256")
+	# Add worker input data
 	wo_submit_json.add_in_data(message)
 
 	# Sign work order
 	private_key = utility.generate_signing_keys()
-
+	wo_submit_json.add_encrypted_request_hash()
+	wo_submit_json.add_requester_signature(private_key)
 	# Submit work order
-	direct_wrapper.init_work_order(config)
-	direct_wrapper.work_order_submit(wo_submit_json, encrypted_session_key, 
-		worker_obj, private_key, session_key, session_iv)
+	logger.info("Work order submit request : %s, \n \n ",
+        wo_submit_json.to_string())
+	req_id += 1
+	direct_jrpc.create_work_order_adaptor(
```

Comment:
direct_jrpc.create_work_order_adaptor() is called many times. Can we create the object once and reuse it. If config needs to be modified add a function update_config() in that class

Reply:
object is created one time only, look at the create_work_order_adaptor(), logic is create the object is not created other wise return already created object.

Reply:
That's actually not good unless you want to implement singleton pattern which I don;t think we need. There could be cases where we need to create multiple adaptor objects. So my suggestion is to change it i.e create_work_order_adaptor() should create new object every time we call it. eco/heart eval client can call only it only once.